daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/processors/dfa/Runtime.scala

```diff
@@ -133,6 +134,17 @@ trait DFAField extends DFA {
   final override def run(r: Registers): Unit = runLoop(r, DFA.EndOfData, StateKind.EndOfData)
 }
 
+object DFADelimiter {
+  private val controlOrWhitespace = "\\p{C}|\\p{Z}".r
+
+  private def containsCtrlOrWS(s: String) = controlOrWhitespace.findFirstMatchIn(s).isDefined
+
+  def strForDiagnostic(s: String) =
+    if (containsCtrlOrWS(s))
```

Comment:
My thought was not about performance. This is only invoked when we're about to cause an error though as a PE it could get backtracked, so performance is an issue. This was mostly to avoid showing a second version of the characters which convert whitespace and control characters to visible things but not annoying the user with that if they aren't present.

Reply:
This has been addressed in latest commit. I elimiinated the duplicate code in Misc for various remappers and they all use sensible non-copying loops now.